test: fix JUnit test method access modifiers and annotations

contrueCT opened the pull request:
<!--
    Licensed to the Apache Software Foundation (ASF) under one or more
    contributor license agreements.  See the NOTICE file distributed with
    this work for additional information regarding copyright ownership.
    The ASF licenses this file to You under the Apache License, Version 2.0
    (the "License"); you may not use this file except in compliance with
    the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0
    
    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
-->
<!-- Please make sure you have read and understood the contributing guidelines -->

- [x] I have read the [CONTRIBUTING.md](https://github.com/apache/incubator-seata/blob/2.x/CONTRIBUTING.md) guidelines.
- [x] I have registered the PR [changes](https://github.com/apache/incubator-seata/tree/2.x/changes).

### Ⅰ. Describe what this PR did

This PR corrects JUnit test method access modifiers and annotation usage:

- Fix `@BeforeEach`/`@AfterEach` method visibility in JUnit 5 tests (private → public)
- Fix `@Rule` field access modifier in JUnit 4 tests (private → public)  
- Remove inappropriate `@Test` annotation from utility methods

These changes ensure proper JUnit framework functionality and test execution.

### Ⅱ. Does this pull request fix one issue?
<!-- If that, add "fixes #xxx" below in the next line, for example, fixes #97. -->


### Ⅲ. Why don't you add test cases (unit test/integration test)? 


### Ⅳ. Describe how to verify it


### Ⅴ. Special notes for reviews

contrueCT committed:
fix: correct JUnit 5 lifecycle method visibility

contrueCT committed:
Fix JUnit 5 lifecycle method visibility in changelog

contrueCT committed:
Fix JUnit 5 lifecycle method visibility in changelog

contrueCT committed:
Merge remote-tracking branch 'origin/patch-1' into 2.x

contrueCT committed:
Merge remote-tracking branch 'origin/patch-2' into 2.x

contrueCT committed:
Update pull request reference in 2.x.md

contrueCT committed:
Update pull request reference in 2.x.md

YongGoose reviewed (changes requested):
Are there similar cases elsewhere? 
If so, it would be good to address all of them in this PR.

contrueCT commented:
> Are there similar cases elsewhere? If so, it would be good to address all of them in this PR.

Got it, I will look for similar cases and update the PR accordingly.

contrueCT committed:
fix: correct JUnit test method access modifiers and annotations

- Fix @BeforeEach/@AfterEach methods visibility in JUnit 5 tests
- Fix @Rule field access modifier from private to public in JUnit 4 tests
- Remove inappropriate @Test annotation from utility method

contrueCT committed:
Merge remote-tracking branch 'origin/2.x' into 2.x

contrueCT commented:
> Are there similar cases elsewhere? If so, it would be good to address all of them in this PR.

Thanks for your suggestion! I’ve checked and fixed all similar cases in the test classes. Could you please take another look when you get a chance?

contrueCT committed:
fix JUnit test method access modifiers and annotations in changelog

YongGoose commented:
> Thanks for your suggestion! I’ve checked and fixed all similar cases in the test classes. Could you please take another look when you get a chance?

Please use JUnit5 instead of JUnit4.

contrueCT committed:
refactor: migrate EtcdRegistryServiceImplTest from JUnit 4 to JUnit 5

contrueCT commented:
Done. Replaced all JUnit4 references with JUnit5. Please take another look when you get a chance.

xingfudeshi committed:
Merge branch '2.x' into 2.x

contrueCT committed:
fix: format code by spotless

contrueCT committed:
Merge remote-tracking branch 'origin/2.x' into 2.x

contrueCT committed:
Merge branch '2.x' into 2.x

contrueCT committed:
Merge branch '2.x' into 2.x

contrueCT commented:
@YongGoose I've updated the PR based on your feedback. Could you please take another look when you have time? Thanks!

xingfudeshi committed:
Merge branch '2.x' into 2.x

contrueCT committed:
fix: correct updateSql assignment in DATE_TYPE and BINARY_TYPE test cases

contrueCT committed:
Merge remote-tracking branch 'origin/2.x' into 2.x

contrueCT commented:
The CI failure in the seata-config-zk module only occurs in the Java 8 build and is unrelated to my changes. The same tests pass successfully in Java 17 and Java 21, indicating a compatibility issue between ZooKeeper client and Java 8 in the CI environment—not a code problem.

YoWuwuuuw commented on discovery/seata-discovery-etcd3/src/test/java/org/apache/seata/discovery/registry/etcd3/EtcdRegistryServiceImplTest.java:
This test class is an integration test that relies on a real etcd service. Because it is marked with `@Disable`, CI will not be triggered. I think it is helpful to include a screenshot of the active triggering test in the description for reference.

slievrly committed:
Merge branch '2.x' into 2.x

contrueCT committed:
Merge branch '2.x' into 2.x

contrueCT commented:
Hi @YongGoose ,  
I've addressed all the feedback you left in your review [https://github.com/apache/incubator-seata/pull/7635#issuecomment-3291173245](url)
Could you please take another look when you have a moment?  
Thanks a lot! 😊

slievrly commented:
@YongGoose Pls take a look.

slievrly committed:
Merge branch '2.x' into 2.x

YongGoose committed:
Merge branch '2.x' into 2.x

YongGoose commented on discovery/seata-discovery-etcd3/src/test/java/org/apache/seata/discovery/registry/etcd3/EtcdRegistryServiceImplTest.java:
```diff
@@ -118,8 +141,8 @@ public void testSubscribe() throws Exception {
         EtcdListener etcdListener = new EtcdListener();
-        registryService.subscribe(CLUSTER_NAME, etcdListener);
-        // 3.delete instance,see if the listener can be notified
+        registryService.subscribe(DEFAULT_TX_GROUP, etcdListener);
```
Just curious, was there any particular reason for using `DEFAULT_TX_GROUP` instead of CLUSTER_NAME?

YongGoose reviewed (approved):
Sorry for the late reply.
LGTM

YongGoose reviewed (changes requested):
Please apply spotless :)

Once you’ve fixed the Spotless issues, please ping me and I’ll give my approval.

YongGoose commented on discovery/seata-discovery-etcd3/src/test/java/org/apache/seata/discovery/registry/etcd3/EtcdRegistryServiceImplTest.java:
```diff
@@ -118,8 +141,8 @@ public void testSubscribe() throws Exception {
         EtcdListener etcdListener = new EtcdListener();
-        registryService.subscribe(CLUSTER_NAME, etcdListener);
-        // 3.delete instance,see if the listener can be notified
+        registryService.subscribe(DEFAULT_TX_GROUP, etcdListener);
```
Just curious, was there any particular reason for using DEFAULT_TX_GROUP instead of CLUSTER_NAME?

contrueCT replied:
Yes! The subscribe() and lookup() methods expect a transaction service group, not an etcd cluster name.
The original code had mismatched key paths:

- Registration: registry-seata-<DEFAULT_TX_GROUP>/...
- Subscription: registry-seata-<CLUSTER_NAME>/...

Even though both values were "default", using DEFAULT_TX_GROUP is semantically correct and matches how Seata actually works - clients look up services by transaction group.

contrueCT committed:
fix:fix the Spotless issues

contrueCT committed:
fix:fix the Spotless issues

contrueCT committed:
Merge branch '2.x' into 2.x

contrueCT commented:
I've fixed the Spotless issues, plaese take a look : ) @YongGoose

YongGoose committed:
Merge branch '2.x' into 2.x

YongGoose reviewed (approved):
LGTM

contrueCT committed:
Merge branch '2.x' into 2.x

YongGoose commented:
@funky-eyes 

I think this pr can be merged now :)

slievrly committed:
Merge branch '2.x' into 2.x

slievrly reviewed (approved):
LGTM
